### PR TITLE
Jest Setup and Added new stage in Jenkins

### PR DESCRIPTION
**Pull request by deepali-hub**
Signed-off-by: Deepali Chavan <deepali.chavan@broadcom.com>

### CODE REVIEW

**Review by Alexandru-Dumitru — CHANGES_REQUESTED** [review]
testMatch for Jest should be `testMatch: ['**/__tests__/**/*-test.ts']` (see hyphen).

We would eventually differentiate the tests like:
Jest:  *-test.ts

Also we should consolidate Jest configs.
Mocha: *.test.ts

Thanks.

**Comment by Alexandru-Dumitru** [comment]
Also, I forgot to mention not use use a template for the commit message. At least not yet, because we don't have this process implemented for this repository. Thanks

File: package.json [hunk 1]
```diff
@@ -299,14 +299,23 @@
...
     "devDependencies": {
         "@types/node": "^12.6.2",
         "tslint": "^5.18.0",
         "tslint-sonarts": "^1.9.0",
-        "typescript": "^3.5.3",
-        "vscode": "^1.1.33"
+        "typescript": "^3.9.4",
+        "vscode": "^1.1.33",
+        "@types/jest": "^26.0.7",
+        "ts-node": "^9.0.0",
```
[the remark this answers is not in the document]
Comment (deepali-hub) [reply package.json]: removed ts-node

**Comment by deepali-hub** [comment]
@Alexandru-Dumitru made all the above mentioned changes. Please review it.

**Review by Alexandru-Dumitru — APPROVED** [review]
Thanks for implementing Jest and addressing my comments. Works well for me

**Pull request merged**